Tests/RudderStackAnalyticsTests/StorageModuleTests/BasicStorageTests.swift

```diff
@@ -0,0 +1,348 @@
+            #expect(dataItem.isClosed == true)
+        }
+    }
+    
+    @Test("given BasicStorage with event batch, when removing specific batch, then batch is deleted")
+    func testEventStorageRemoveSpecificBatch() async {
+        await storage.write(event: sampleEventJson)
+        await storage.rollover()
+        
+        var result = await storage.read()
+        #expect(result.dataItems.count == 1)
+        
+        guard let batchReference = result.dataItems.first?.reference else {
+            Issue.record("Expected at least one data item")
+            return
+        }
+        let removed = await storage.remove(batchReference: batchReference)
+        
+        #expect(removed == true)
+        
+        result = await storage.read()
+        #expect(result.dataItems.isEmpty)
+    }
+    
+    @Test("given BasicStorage, when removing non-existent batch, then operation returns false")
+    func testEventStorageRemoveNonExistentBatch() async {
+        let fakeBatchReference = "non-existent-batch-reference"
+        let removed = await storage.remove(batchReference: fakeBatchReference)
+        
+        #expect(removed == false)
+    }
+    
+    @Test("given BasicStorage, when calling rollover without events, then no batch is created")
+    func testEventStorageRolloverWithoutEvents() async {
+        await storage.rollover()
+        
+        let result = await storage.read()
+        #expect(result.dataItems.isEmpty)
+    }
+    
+    // MARK: - Combined Storage Operations Tests
+    
+    @Test("given BasicStorage, when using both key-value and event storage, then both work independently")
+    func testCombinedStorageOperations() async {
+        // Store key-value data
+        let key = "combined_test_key"
+        let value = "combined_test_value"
+        storage.write(value: value, key: key)
+        
+        // Store event data
+        await storage.write(event: sampleEventJson)
+        await storage.rollover()
+        
+        // Verify key-value storage
+        let retrievedValue: String? = storage.read(key: key)
+        #expect(retrievedValue == value)
+        
+        // Verify event storage
+        let eventResult = await storage.read()
+        #expect(eventResult.dataItems.count == 1)
+        #expect(eventResult.dataItems.first?.batch.contains("Test Event") == true)
+    }
+    
+    @Test("given BasicStorage with mixed data, when calling removeAll, then all data is cleared")
+    func testRemoveAllData() async {
+        // Store key-value data
+        storage.write(value: "test_value", key: "test_key")
+        
+        // Store event data
+        await storage.write(event: sampleEventJson)
+        await storage.rollover()
+        
+        // Verify data exists
+        let keyValue: String? = storage.read(key: "test_key")
+        #expect(keyValue == "test_value")
+        
+        let eventResult = await storage.read()
+        #expect(eventResult.dataItems.count == 1)
```

Comment:
Let's try to avoid verifying anything before we write the "when" phase.

Reply:
Our aim should be to align our test cases with the `given.. when.. then..` statement. 

Removing the early `verify` statement makes our test cleaner. 

As said earlier, if you feel they are needed, feel free to add a new test case with those lines.